setup.cfg

```
@@ -16,4 +16,19 @@ profile = black
 combine_as_imports = True
 
 [tool:pytest]
-addopts = --cov-report= --cov=starlette --cov=tests -rxXs
+addopts =
+  --cov-report=term-missing:skip-covered
+  --cov=starlette
+  --cov=tests
+  -rxXs
+  --strict-config
+  --strict-markers
+  --cov-config tests/.ignore_lifespan
+  --cov-fail-under=100
+xfail_strict=True
+filterwarnings=
+    error
```

and there's an unclosed eventloop, and unclosed socket:

```
Exception ignored in: <function BaseEventLoop.__del__ at 0x7f6cfc2e6f70>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/asyncio/base_events.py", line 654, in __del__
    _warn(f"unclosed event loop {self!r}", ResourceWarning, source=self)
ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
Exception ignored in: <socket.socket fd=21, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/_pytest/logging.py", line 680, in _runtest_for
    with catching_logs(
ResourceWarning: unclosed <socket.socket fd=21, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0>
Exception ignored in: <socket.socket fd=22, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/_pytest/logging.py", line 680, in _runtest_for
    with catching_logs(
ResourceWarning: unclosed <socket.socket fd=22, family=AddressFamily.AF_UNIX, type=SocketKind.SOCK_STREAM, proto=0>
```

---

https://github.com/pytest-dev/pytest/issues/5299

---

Hrm..we should probably fix these 🤔 curious that they all seem to be warnings for `family=AddressFamily.AF_UNIX`

---

@JayH5 I think I've got a fix for the requests session not being closed

---

@JayH5 nope, it made it worse, it looks like a lot of code isn't expecting to be closed properly - eg the _ASGIAdaptor needs to override `.close(`


https://github.com/encode/starlette/issues/1050

---

Sorry, would you mind explaining the `error` line? What is that ignoring? Removing it doesn't seem to make a difference for me when I try locally.

I also don't seem to get that ResourceWarning every time I run the tests 😕

---

the `error` like is the same as running with `pytest -Werror`, it un-ignores warnings because the default is to ignore them

the ResourceWarning is intermittent because it's GC related

---

Thanks for the explanation.

Would it be possible to add a code comment explaining this briefly? Otherwise it could easily be read as "ignore errors".
```suggestion
    # Turn warnings that aren't filtered into exceptions
    error
```